gems/smithy/spec/support/client_helper.rb

```diff
@@ -0,0 +1,182 @@
+
+module ClientHelper
+  class << self
+    def sample_shapes
+      {
+        'smithy.ruby.tests#SampleClient' => {
+          'type' => 'service',
+          'operations' => [
+            { 'target' => 'smithy.ruby.tests#Operation' }
+          ]
+        },
+        'smithy.ruby.tests#Operation' => {
+          'type' => 'operation',
+          'input' => { 'target' => 'smithy.ruby.tests#Structure' },
+          'output' => { 'target' => 'smithy.ruby.tests#Structure' }
+        },
+        'smithy.ruby.tests#Enum' => {
+          'type' => 'enum',
+          'members' => {
+            'member' => {
+              'target' => 'smithy.api#Unit',
+              'traits' => { 'smithy.api#enumValue' => 'value' }
+            }
+          }
+        },
+        'smithy.ruby.tests#intEnum' => {
+          'type' => 'intEnum',
+          'members' => {
+            'member' => {
+              'target' => 'smithy.api#Unit',
+              'traits' => { 'smithy.api#enumValue' => 1 }
+            }
+          }
+        },
+        'smithy.ruby.tests#List' => {
+          'type' => 'list',
+          'member' => { 'target' => 'smithy.api#String' }
+        },
+        'smithy.ruby.tests#Map' => {
+          'type' => 'map',
+          'key' => { 'target' => 'smithy.api#String' },
+          'value' => { 'target' => 'smithy.api#String' }
+        },
+        'smithy.ruby.tests#Structure' => {
+          'type' => 'structure',
+          'members' => {
+            'bigDecimal' => { 'target' => 'smithy.api#BigDecimal' },
+            'bigInteger' => { 'target' => 'smithy.api#BigInteger' },
+            'blob' => { 'target' => 'smithy.api#Blob' },
+            'boolean' => { 'target' => 'smithy.api#Boolean' },
+            'byte' => { 'target' => 'smithy.api#Byte' },
+            'document' => { 'target' => 'smithy.api#Document' },
+            'double' => { 'target' => 'smithy.api#Double' },
+            'enum' => { 'target' => 'smithy.ruby.tests#Enum' },
+            'float' => { 'target' => 'smithy.api#Float' },
+            'intEnum' => { 'target' => 'smithy.ruby.tests#intEnum' },
+            'integer' => { 'target' => 'smithy.api#Integer' },
+            'list' => { 'target' => 'smithy.ruby.tests#List' },
+            'long' => { 'target' => 'smithy.api#Long' },
+            'map' => { 'target' => 'smithy.ruby.tests#Map' },
+            'short' => { 'target' => 'smithy.api#Short' },
+            'string' => { 'target' => 'smithy.api#String' },
+            'structure' => { 'target' => 'smithy.ruby.tests#Structure' },
+            'timestamp' => { 'target' => 'smithy.api#Timestamp' },
+            'union' => { 'target' => 'smithy.ruby.tests#Union' }
+          }
+        },
+        'smithy.ruby.tests#Union' => {
+          'type' => 'union',
+          'members' => {
+            'string' => { 'target' => 'smithy.api#String' }
+          }
+        }
+      }
+    end
+
+    # @param [Array<String>] module_names A list of module names for the
```

Comment:
Is there a reason we need this as an array insead of as just a full name eg "Company::Weather"

Reply:
I am going for full name. The docs are not correct (sloppy editing)